### DISCUSSION TITLE
mandatory change, jsx was set to preserve

### DISCUSSION THREAD

**Discussion by CaioMS2000** [vercel/next.js#54456] [Help]
### Summary

When I start development mode(next dev) Next automatcally says:
```
We detected TypeScript in your project and reconfigured your tsconfig.json file for you. Strict-mode is set to false by default.

The following mandatory changes were made to your tsconfig.json:

        - jsx was set to preserve (next.js implements its own optimized jsx transform)
```
But it ruins  intellisense for Typescript in vscode, it doesn't recognize react resourses(like useEffect, useState, etc) and I always have to manually set **

> "jsx": "react-jsx"

**. What should I do?

### Additional information

```js
tsconfig.json:
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "bundler",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    // "jsx": "react-jsx",
    "incremental": true,
    "strictPropertyInitialization": false,
    "plugins": [
      {
        "name": "next"
      }
    ],
    "paths": {
      "@/*": [
        "./*"
      ]
    }
  },
  "include": [
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx",
    ".next/types/**/*.ts"
  ],
  "exclude": [
    "node_modules"
  ]
}
```

**Comment by devjiwonchoi** [comment]
> it ruins intellisense for Typescript in vscode

This shouldn't be true. Which intellisense are you talking about?

BTW, this is the default tsconfig when `create-next-app` try this if you still have the issue.

```ts
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "bundler",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "plugins": [
      {
        "name": "next"
      }
    ],
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
  "exclude": ["node_modules"]
}
```

**Reply by CaioMS2000** [reply]
I'm talking about the built-in intellisense. I have the "ES7+ React/Redux/React-Native snippets", "ESLint" and "JavaScript and TypeScript Nightly" extensions installed, I don't know if it interferes, but this problem started very recently when I started using one of the latest versions of NextJS(currently I'm using 13.4.13).

**Reply by devjiwonchoi** [reply]
Both extensions you've mentioned are not vscode built-ins.
As the log has stated, `jsx: "preserve"` is mandatory.

> But it ruins intellisense for Typescript in vscode, it doesn't recognize react resourses(like useEffect, useState, etc)

Could you share any logs or lints of the statement above?

**Reply by CaioMS2000** [reply]
I know they are not built-ins, I mentioned them just in case they could interfer. But I don't know how could I get logs of it. Something weird I noticed yesterday, it sometimes works correctly and sometimes doesn't. Any chance of my editor just had a occasional bug?
